dbstats_go1.11.go

```diff
@@ -0,0 +1,46 @@
+// +build go1.11
+
+package ocsql
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"go.opencensus.io/stats"
+)
+
+// RecordStats records database statistics for provided sql.DB at the provided
+// interval.
+func RecordStats(db *sql.DB, interval time.Duration) (fnStop func()) {
+	var (
+		ctx    = context.Background()
+		ticker = time.NewTicker(interval)
+		done   = make(chan struct{})
+	)
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				dbStats := db.Stats()
+				stats.Record(ctx,
+					MeasureOpenConnections.M(int64(dbStats.OpenConnections)),
+					MeasureIdleConnections.M(int64(dbStats.Idle)),
+					MeasureActiveConnections.M(int64(dbStats.InUse)),
+					MeasureWaitCount.M(dbStats.WaitCount),
+					MeasureWaitDuration.M(float64(dbStats.WaitDuration.Nanoseconds())/1e6),
+					MeasureIdleClosed.M(dbStats.MaxIdleClosed),
+					MeasureLifetimeClosed.M(dbStats.MaxLifetimeClosed),
+				)
+			case <-done:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+
+	return func() {
+		close(done)
```

Comment:
Actually what we can do is ensure that this ever gets called once otherwise calling this again will cause a panic and some folks might misuse it.
```go
var closeOnce sync.Once

return func() {
    closeOnce.Do(func() {
      close(done)
    })
}()
```

Reply:
yep, will do...